### PR TITLE
Style TD in-game chat with faction colors

### PR DESCRIPTION
**Pull request by dragunoff**
Style the in-game chat for GDI with the faction chrome that already exists. The green for GDI was originally made by shifting the hue but overall it's a little brighter than the default red. This produced poorer contrast between the background for the chat and the text so for this PR I have tweaked the `pnael-gray-gdi` background by slightly increasing the opacity. I will open a PR at https://github.com/OpenRA/ArtSrc/ with the changes to `chrome.svg`.

faction-chat-gdi
_GDI in-game chrome_

faction-chat-nod
_Nod in-game chrome_

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/Logic/Ingame/AddFactionSuffixLogic.cs [hunk 1]
```diff
@@ -29,13 +29,22 @@ public AddFactionSuffixLogic(Widget widget, World world)
 				((ImageWidget)widget).ImageCollection += suffix;
 			else if (widget is BackgroundWidget)
 				((BackgroundWidget)widget).Background += suffix;
+			else if (widget is TextFieldWidget)
+				((TextFieldWidget)widget).Background += suffix;
+			else if (widget is ScrollPanelWidget)
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/Logic/Ingame/AddFactionSuffixLogic.cs]: Can we please switch these all over to the newer pattern matching syntax while we're here?
```csharp
else if (widget is ScrollPanelWidget spw)
{
	spw.Button += suffix;
	spw.Background += suffix;
	spw.ScrollBarBackground += suffix;
	spw.Decorations += suffix;
}
```

**Comment by dragunoff** [comment]
Update: Switched to pattern matching syntax.

**Review by pchote — DISMISSED** [review]
LGTM. Can we also do the perf graph background (doesn't need to be in this PR)?

**Comment by dragunoff** [comment]
Update: Added faction suffix to in-game perf graph background.

**Review by pchote — APPROVED** [review]

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/dd2f05a00e66dc23dd8a7e713755dec8a5f01eb8...79aa4fc760f8533c98cbf26380752fe3f92b3ad7)

**Pull request merged**